src/pseudo/load_psp.jl

```diff
@@ -15,7 +15,7 @@ function load_psp(key::AbstractString)
         parser = parse_hgh_file
```

Comment:
Also make this a constructor and rename `parser` to PspType or something like that.

Reply:
> The issue is you are not just imposing to have the field, but also imposing something on the structure of the field, which can imply built-in assumptions about the data layout

You can still overload getproperty if you want to do exotic stuff

> Who says you need two different names? f(x) = x.f works fine.

But then you can't do in a function `f = f(x)`. Also since it's a namespaced function, you probably want to avoid short identifiers like Zion, and either you give up short identifiers, or you have two different names (like here)

Reply:
> You can still overload getproperty if you want to do exotic stuff

Well ... or not impose so much in the first place don't hide the exotic stuff in getproperty? Also it can be a bit tricky to make getproperty type-stable.

Reply:
Eg on this particular instance, is it really unreasonable to require a pseudo structure to have a field that has the number of valence electrons? By exotic I meant have a struct that doesn't have this as a field (maybe because the layout is imposed by an external tool or what, although encapsulation is a better solution). In that unlikely case, getproperty is an option.

Reply:
Also we've been fine requiring lmax, h, identifier and description to be fields, why is this one different? Or do you want accessor functions for lmax and h? If so, how do we name them?

Reply:
Sure I get your point. I mean there is a reason why I never went ahead and changed this (also I removed one required field at some point) :smile:.

But talking about `lmax` that's an interesting case. Because that one you could e.g. deduce from the `h`. I've always wondered why we have not done that actually. Just have a function `lmax` instead?

Reply:
Right but having a function named lmax is terrible, because it's too short without context and then we can't use it as a variable name. All of these should just be required fields and that's it, I don't understand your objection (other than a generic "fields bad, functions good" that makes sense in C++/java but not in julia)

Reply:
Maybe we leave this for a future PR on pseudo interfaces / for when and if we move the pseudos out to another package?

Reply:
> All of these should just be required fields and that's it, I don't understand your objection (other than a generic "fields bad, functions good" that makes sense in C++/java but not in julia)

Oh sorry, maybe we misunderstood. I'm not saying required fields are bad in general. I'm only saying they cement more structure about how state is to be stored and manipulated than requiring functions does. Therefore one should not be too eager with that. And regarding the `lmax`, it just bothers me that this is duplicated information to the `h`, thus just enforcing the `h` should be enough. But yeah a function named `lmax` is probably not great either, I agree on that.

And coming back to the valence electron case: The point is we already have a `charge_ionic` function, which we need to dispatch on the elements, so why not use it here? Or do you suggest to give up on `charge_ionic(element)` as 

@azadoks I'm not suggesting to change this in this PR in case that came across (and I don't think @antoine-levitt suggests to make any changes right now either). Yes we should have the detailed discussion when we think about the general interface.